### ISSUE TITLE
dependency require-from-string v2.0.2 is broken

### ISSUE THREAD

**Issue by pabo** [davidtheclark/cosmiconfig#123]
oops - it looks like this is just an issue on the private registry I'm using. disregard.

**Comment by davidlukerice** [comment]
What did you do to fix the issue? We're also behind a private registry and are having trouble with this.

**Comment by pabo** [comment]
We haven't fixed it yet - I was just able to isolate the issue. If you compare the tarball that is provided by the public registry to that provided by your private registry, you'll notice it's different and has a different sha. 

```
$ curl -s https://registry.npmjs.org/require-from-string | jq '.versions["2.0.2"].dist'
{
  "integrity": "sha512-Xf0nWe6RseziFMu+Ap9biiUbmplq6S9/p+7w7YXP/JBHhrUDDUhwa+vANyubuqfZWTveU//DYVGsDG7RKL/vEw==",
  "shasum": "89a7fdd938261267318eafe14f9c32e598c36909",
  "tarball": "https://registry.npmjs.org/require-from-string/-/require-from-string-2.0.2.tgz",
  "fileCount": 4,
  "unpackedSize": 3422
}

$ curl -s https://registry.npmjs.org/require-from-string/-/require-from-string-2.0.2.tgz | shasum
89a7fdd938261267318eafe14f9c32e598c36909  -
```

```
$ curl -s https://registry......./require-from-string | jq '.versions["2.0.2"].dist'
{
  "integrity": "sha512-Xf0nWe6RseziFMu+Ap9biiUbmplq6S9/p+7w7YXP/JBHhrUDDUhwa+vANyubuqfZWTveU//DYVGsDG7RKL/vEw==",
  "shasum": "89a7fdd938261267318eafe14f9c32e598c36909",
  "tarball": "https://registry....../r/require-from-string/_attachments/require-from-string-2.0.2.tgz",
  "fileCount": 4,
  "unpackedSize": 3422
}

$ curl -s https://registry....../r/require-from-string/_attachments/require-from-string-2.0.2.tgz | shasum
9fce58558db06add0e02da5771ac3b6cd93881a9  -
```

**Comment by pabo** [comment]
@davidlukerice FYI we were finally able to fix this issue on our private registry. It looks like somehow the require-from-string tarball was changed at some point. `require-from-string-2.0.2.tgz`

The public NPM registry does not guarantee sameness for tarballs. So if your private registry fetched the tarball when it was first published, but then the tarball was since updated, your private registry will be out of date. It seems like a fix would be to have the private registry detect that the sha has changed and so fetch the new tarball to match.

We just manually uploaded the newer tarball to our private repo.
